### PR TITLE
Add text defining storage root and object root 'extensions' directories

### PR DESCRIPTION
**Pull request by awoods**
Resolves: https://github.com/OCFL/spec/issues/403

### CODE REVIEW

File: draft/spec/index.html [hunk 1]
```diff
@@ -954,10 +979,18 @@ <h2>Storage Hierarchies</h2>
...
 
-    <section id="root-extensions">
-        <h2>Extensions</h2>
+    <section id="storage-root-extensions">
+        <h2>Storage Root Extensions</h2>
         <p>
             The behavior of the storage root may be extended to support features from other specifications.
+        </p>
+        <p>
+            The base directory of an OCFL Object MAY contain a directory named <code>extensions</code> for the purposes
```
Comment (zimeon) [inline draft/spec/index.html]: s/OCFL Object/OCFL Storage Root/

File: draft/spec/index.html [hunk 2]
```diff
@@ -954,10 +979,18 @@ <h2>Storage Hierarchies</h2>
...
+            of extending the functionality of an OCFL Object. The storage root <code>extensions</code> directory MUST
```
Comment (zimeon) [inline draft/spec/index.html]: s/OCFL Object/OCFL Storage Root/

File: draft/spec/index.html [hunk 3]
```diff
@@ -1514,6 +1547,27 @@ <h2>Moab in an OCFL Object</h2>
...
+            The following example OCFL Object has an extension containing custom content. The OCFL Object itself
+            remains valid.
+        </p>
+<pre>
+[object root]
+    ├── 0=ocfl_object_1.0
+    ├── inventory.json
+    ├── inventory.json.sha512
+    ├── extensions
+    │   └── 0011-my-extension
```
Comment (zimeon) [inline draft/spec/index.html]: Do we need to create an example extension, maybe `9999-example-extension` or such in order for this to be valid?

File: draft/spec/index.html [hunk 4]
```diff
@@ -861,7 +861,32 @@ <h2>Logs Directory</h2>
...
 
+    <section id="object-extensions">
+        <h2>Object Extensions</h2>
+        <p>
+            The base directory of an OCFL Object MAY contain a directory named <code>extensions</code> for the purposes
+            of extending the functionality of an OCFL Object. The <code>extensions</code> directory MUST NOT contain
+            any files, and no sub-directories other than extension sub-directories. Extension sub-directories MUST be
+            named according to a registered extension name. The specific structure and function of the extension, as
+            well as a declaration of the registered extension name MUST be defined in one of the following locations:
+            <ul>
```
Comment (zimeon) [inline draft/spec/index.html]: Need to close the `<p>` block before opening `<ul>`, this is the reason for the travis error (and, of course, remove `<\p>` after)

File: draft/spec/index.html [hunk 5]
```diff
@@ -861,7 +861,32 @@ <h2>Logs Directory</h2>
...
             </p>
         </blockquote>
+    </section>
 
+    <section id="object-extensions">
+        <h2>Object Extensions</h2>
+        <p>
+            The base directory of an OCFL Object MAY contain a directory named <code>extensions</code> for the purposes
+            of extending the functionality of an OCFL Object. The <code>extensions</code> directory MUST NOT contain
+            any files, and no sub-directories other than extension sub-directories. Extension sub-directories MUST be
```
Comment (zimeon) [inline draft/spec/index.html]: Per https://github.com/OCFL/spec/wiki/2020.01.21-Editors-Meeting I think agreement was that:
  * "Extensions SHOULD be a defined/registered"
  * "Extensions MUST be in their own sub-directory"

**Comment by zimeon** [comment]
A few minor issues above but otherwise I think this looks great!

File: draft/spec/index.html (hunk 3 above)
Comment (rosy1280) [reply draft/spec/index.html]: we have an example one `0000-example-extension` see: https://github.com/OCFL/extensions/blob/master/docs/0000-example-extension.md

File: draft/spec/index.html [hunk 6]
```diff
@@ -861,7 +861,32 @@ <h2>Logs Directory</h2>
...
+            named according to a registered extension name. The specific structure and function of the extension, as
+            well as a declaration of the registered extension name MUST be defined in one of the following locations:
+            <ul>
+              <li>
+                The <a href="https://ocfl.github.io/extensions/">OCFL Extensions repository</a>
+              </li>
+              <li>
+                The Storage Root as a plain text document
```
Comment (ahankinson) [inline draft/spec/index.html]: I can see this leading to confusion that object extensions can be put in the storage root `extensions` directory, not directly in the storage root itself.

File: draft/spec/index.html (hunk 3 above)
Comment (zimeon) [reply draft/spec/index.html]: Thanks @rosy1280 , I'd forgotten that. @awoods change looks good so will resolve

File: draft/spec/index.html [hunk 7]
```diff
@@ -861,7 +861,32 @@ <h2>Logs Directory</h2>
...
+          </li>
+          <li>
+            The Storage Root, as a plain text document directly in the Storage Root
+          </li>
+        </ul>
+        <blockquote class="informative">
+          <p>
+            Non-normative note: Extension sub-directories should use the same name as a registered extension
+            in order to both avoid the possiblity of an extension sub-directory colliding with the name of another
+            registered extension as well as to facilitate the recognition of extensions by OCFL consumer clients.
```
Comment (ahankinson) [inline draft/spec/index.html]: remove 'consumer'? We don't really define what that is.

File: draft/spec/index.html [hunk 8]
```diff
@@ -1514,6 +1547,27 @@ <h2>Moab in an OCFL Object</h2>
 }
 </pre>
 </section>
+    <section id="example-extended-object">
```
Comment (ahankinson) [inline draft/spec/index.html]: Perhaps we need an example Storage Root layout to clarify the extensions mechanism there?

File: draft/spec/index.html (hunk 7 above)
Comment (zimeon) [reply draft/spec/index.html]: agree to drop "consumer"

File: draft/spec/index.html (hunk 8 above)
Comment (zimeon) [reply draft/spec/index.html]: :+1:

**Pull request merged**